### PR TITLE
Improve rendering of `AttachmentThumbnail` when thumbnailSource is not provided

### LINKED ISSUE #1741: Images received from Telegram aren't shown in reply previews
### Steps to reproduce

1. Find a message with image downloaded from Telegram and sent as-is, either:
- bridged via https://github.com/mautrix/telegram, or
- saved from Telegram web client and sent without compression, e.g. using Nheko 0.11.3
2. Attempt replying to the message, or find an existing reply in timeline

#### What did you expect?
There's a preview of the image that's being replied to, both in message input box and in existing replies in room timeline.
#### What happened instead?
https://github.com/vector-im/element-x-android/assets/46846000/fb8fce4d-707a-40b4-bde6-e1fae3f41d60

In this video, first image is a copy that has been downloaded from Telegram and resent to this room using `Photo & Video Library` option, which has compressed it slightly (`Attachment` also modifies the images, a bit unexpected). Second image has been sent untouched, using Nheko. First one is visible in reply preview and in reply on timeline, meanwhile the second - as originally received in another room via Telegram bridge - does not appear in either.

In case you need those files: [images.zip](https://github.com/vector-im/element-x-android/files/13261034/images.zip)

Some clueless sleuthing:
```
$ ls -l
-rw-r--r-- 1 przemek przemek 49683 11-05 23:44 bad.jpeg
-rw-r--r-- 1 przemek przemek 49773 11-05 23:44 good.jpeg
$ xxd bad.jpeg | head
00000000: ffd8 ffe0 0010 4a46 4946 0001 0101 0060  ......JFIF.....`
00000010: 0060 0000 ffdb 0043 0004 0303 0403 0304  .`.....C........
00000020: 0403 0405 0404 0506 0a07 0606 0606 0d09  ................
00000030: 0a08 0a0f 0d10 100f 0d0f 0e11 1318 1411  ................
00000040: 1217 120e 0f15 1c15 1719 191b 1b1b 1014  ................
00000050: 1d1f 1d1a 1f18 1a1b 1aff db00 4301 0405  ............C...
00000060: 0506 0506 0c07 070c 1a11 0f11 1a1a 1a1a  ................
00000070: 1a1a 1a1a 1a1a 1a1a 1a1a 1a1a 1a1a 1a1a  ................
00000080: 1a1a 1a1a 1a1a 1a1a 1a1a 1a1a 1a1a 1a1a  ................
00000090: 1a1a 1a1a 1a1a 1a1a 1a1a 1a1a 1a1a ffc2  ................
$ xxd good.jpeg | head
00000000: ffd8 ffe1 0058 4578 6966 0000 4d4d 002a  .....XExif..MM.*
00000010: 0000 0008 0004 0100 0004 0000 0001 0000  ................
00000020: 0252 0101 0004 0000 0001 0000 01f3 8769  .R.............i
00000030: 0004 0000 0001 0000 003e 0112 0004 0000  .........>......
00000040: 0001 0000 0000 0000 0000 0001 9208 0004  ................
00000050: 0000 0001 0000 0000 0000 0000 ffe0 0010  ................
00000060: 4a46 4946 0001 0101 0060 0060 0000 ffdb  JFIF.....`.`....
00000070: 0043 0004 0303 0403 0304 0403 0405 0404  .C..............
00000080: 0506 0a07 0606 0606 0d09 0a08 0a0f 0d10  ................
00000090: 100f 0d0f 0e11 1318 1411 1217 120e 0f15  ................
```

Sliding to reply to either of those images gives me the same logs:
```
2023-11-05 23:43:22.971 21302-21302 Compose Focus           io.element.android.x.debug           D  Owner FocusChanged(true)
2023-11-05 23:43:22.972 21302-21302 InputMethodManager      io.element.android.x.debug           D  showSoftInput() view=androidx.compose.ui.platform.AndroidComposeView{4d14209 VFED..... .F....ID 0,0-1080,2160 aid=1073741826} flags=1 reason=SHOW_SOFT_INPUT
2023-11-05 23:43:23.006 21302-21302 Compose Focus           io.element.android.x.debug           D  Owner FocusChanged(false)
2023-11-05 23:43:23.051 21302-21302 AssistStructure         io.element.android.x.debug           I  Flattened final assist data: 1100 bytes, containing 1 windows, 6 views
2023-11-05 23:43:23.058 21302-21302 InsetsController        io.element.android.x.debug           D  show(ime(), fromIme=true)
2023-11-05 23:43:23.058 21302-21302 InsetsController        io.element.android.x.debug           D  show(ime(), fromIme=true)
2023-11-05 23:43:23.058 21302-21302 InsetsController        io.element.android.x.debug           D  show(ime(), fromIme=true)
```

Videos received from Telegram are unaffected.

### Your phone model

Google Pixel 3

### Operating system version

Android 12

### Application version and app store

fe69db3, GHA debug build

### Homeserver

Synapse 1.95.1

### Will you send logs?

Yes

### Are you willing to provide a PR?

No

### PR DESCRIPTION
**Pull request by bmarty**
## Type of change

- [ ] Feature
- [x] Bugfix
- [ ] Technical
- [ ] Other :

## Content

Use the blurHash if available, when the thumbnailSource is null, and if not available render the image icon for image file.

When looking at #1741, I noticed that sometimes `AttachmentThumbnailInfo.thumbnailSource` is null, but `AttachmentThumbnailInfo.blurHash` is present. This PR uses that data to improve the rendering.

## Motivation and context

Better UX

Closes #1741 

## Screenshots / GIFs

|Before|After first commit (ignore text and theme)|After second commit|
|-|-|-|
|image|image|image|
|image||image|

## Tests

- Send an image to a room
- Reply to this message

Before the image was not rendered in the text composer, and the reply message did not contain the image in the thumbnail. Now it's there for both cases.

## Tested devices

- [ ] Physical
- [ ] Emulator
- OS version(s):

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-x-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR

### CODE REVIEW

File: tests/uitests/src/test/snapshots/images/ui_S_t[l.matrix.ui.components_UnsavedAvatar_null_UnsavedAvatar-Night-9_11_null,NEXUS_5,1.0,en].png
Comment (bmarty) [inline tests/uitests/src/test/snapshots/images/ui_S_t[l.matrix.ui.components_UnsavedAvatar_null_UnsavedAvatar-Night-9_11_null,NEXUS_5,1.0,en].png:1]: I do not know why all those files have been renamed :/

File: tests/uitests/src/test/snapshots/images/ui_S_t[f.messages.impl.actionlist_SheetContent_null_SheetContent-Day-1_1_null_8,NEXUS_5,1.0,en].png
Comment (jmartinesp) [inline tests/uitests/src/test/snapshots/images/ui_S_t[f.messages.impl.actionlist_SheetContent_null_SheetContent-Day-1_1_null_8,NEXUS_5,1.0,en].png:1]: This icon changed, weird.

File: libraries/matrixui/src/main/kotlin/io/element/android/libraries/matrix/ui/components/AttachmentThumbnail.kt [hunk 1]
```diff
@@ -89,11 +110,11 @@ fun AttachmentThumbnail(
                     )
                 }
                 AttachmentThumbnailType.Location -> {
-                    PinIcon(
-                        modifier = Modifier.fillMaxSize()
+                    Icon(
+                        resourceId = R.drawable.ic_september_location,
+                        contentDescription = info.textContent,
```
Comment (jmartinesp) [inline libraries/matrixui/src/main/kotlin/io/element/android/libraries/matrix/ui/components/AttachmentThumbnail.kt:115]: Was this change intended? Is this based on any designs?

File: tests/uitests/src/test/snapshots/images/ui_S_t[f.messages.impl.actionlist_SheetContent_null_SheetContent-Day-1_1_null_8,NEXUS_5,1.0,en].png
Comment (bmarty) [reply tests/uitests/src/test/snapshots/images/ui_S_t[f.messages.impl.actionlist_SheetContent_null_SheetContent-Day-1_1_null_8,NEXUS_5,1.0,en].png:1]: No, it's expected, I changed it for coherency with the other icons.

See the preview in AS:

|Before|After|
|-|-|
|image|image|

File: libraries/matrixui/src/main/kotlin/io/element/android/libraries/matrix/ui/components/AttachmentThumbnail.kt (hunk 1 above)
Comment (bmarty) [reply libraries/matrixui/src/main/kotlin/io/element/android/libraries/matrix/ui/components/AttachmentThumbnail.kt:115]: Good spot, and I did not advertise it (sorry!). Rational is coherency, see my comment https://github.com/vector-im/element-x-android/pull/1774#discussion_r1387642675

File: tests/uitests/src/test/snapshots/images/ui_S_t[f.messages.impl.actionlist_SheetContent_null_SheetContent-Day-1_1_null_8,NEXUS_5,1.0,en].png
Comment (jmartinesp) [reply tests/uitests/src/test/snapshots/images/ui_S_t[f.messages.impl.actionlist_SheetContent_null_SheetContent-Day-1_1_null_8,NEXUS_5,1.0,en].png:1]: I'd ask someone from the design team first, as Figma designs are using the previous icon: https://www.figma.com/file/0MMNu7cTOzLOlWb7ctTkv3/Element-X?type=design&node-id=3504%3A147257&mode=design&t=9p86HyEaujXj0rzB-1

Comment (bmarty) [reply tests/uitests/src/test/snapshots/images/ui_S_t[f.messages.impl.actionlist_SheetContent_null_SheetContent-Day-1_1_null_8,NEXUS_5,1.0,en].png:1]: @callumu / @amshakal can you confirm this change please?

File: libraries/matrixui/src/main/kotlin/io/element/android/libraries/matrix/ui/components/AttachmentThumbnail.kt (hunk 1 above)
Comment (bmarty) [reply libraries/matrixui/src/main/kotlin/io/element/android/libraries/matrix/ui/components/AttachmentThumbnail.kt:115]: I will revert the change to unblock the PR, since I would like the fix to be in the next release.

**Review by jmartinesp — APPROVED** [review]

File: tests/uitests/src/test/snapshots/images/ui_S_t[f.messages.impl.actionlist_SheetContent_null_SheetContent-Day-1_1_null_8,NEXUS_5,1.0,en].png
Comment (bmarty) [reply tests/uitests/src/test/snapshots/images/ui_S_t[f.messages.impl.actionlist_SheetContent_null_SheetContent-Day-1_1_null_8,NEXUS_5,1.0,en].png:1]: Thanks for the feedback. I guess it will be more efficient to look at all the icons we have with you and replace the outdated one(s). Also sometimes we add an extra [rotation](https://github.com/vector-im/element-x-android/blob/develop/features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/media/local/LocalMediaView.kt#L244), and this is maybe not expected anymore...

Comment (callumu) [reply tests/uitests/src/test/snapshots/images/ui_S_t[f.messages.impl.actionlist_SheetContent_null_SheetContent-Day-1_1_null_8,NEXUS_5,1.0,en].png:1]: > Thanks for the feedback. I guess it will be more efficient to look at all the icons we have with you and replace the outdated one(s). Also sometimes we add an extra [rotation](https://github.com/vector-im/element-x-android/blob/develop/features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/media/local/LocalMediaView.kt#L244), and this is maybe not expected anymore...

Sure, I'm happy to have a call and discuss this if it's easier?

I feel like me providing some as SVGs and via Compound might be confusing / not the best way to do it
